### PR TITLE
updating content inheritance on page builder

### PR DESCRIPTION
**Pull request by aachavarria**

### CODE REVIEW

File: ui/app/src/models/ContentInstance.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export interface ContentInstanceSystemProps {
   dateCreated: string;
   dateModified: string;
   contentTypeId: string;
-  sourceMap?: { [path: string]: string }; // path: contentTypeId
+  sourceMap?: { [fieldId: string]: string }; // fieldId: path
```
Comment (rart) [inline ui/app/src/models/ContentInstance.ts:25]: I’m thinking we could have both things here. When parsing content the sourceMap would be populated with both { fieldId: path, path: contentTypeId }.

File: ui/guest/src/classes/ICERegistry.ts [hunk 2]
```diff
@@ -324,10 +324,18 @@ export function runValidation(iceId: number, validationId: ValidationKeys, args?
...
+  let model = contentController.getCachedModel(record.modelId);
+  let contentTypeId = Model.getContentTypeId(model);
+  let contentType = contentController.getCachedContentType(contentTypeId);
+  let field = record.fieldId ? contentTypeUtils.getField(contentType, record.fieldId) : null;
+
+  if (!field && record.fieldId && model.craftercms.sourceMap) {
+    model = contentController.getContentInstanceByPath(model.craftercms.sourceMap[record.fieldId]);
+    contentTypeId = Model.getContentTypeId(model);
+    contentType = contentController.getCachedContentType(contentTypeId);
+    field = record.fieldId ? contentTypeUtils.getField(contentType, record.fieldId) : null;
```
Comment (rart) [inline ui/guest/src/classes/ICERegistry.ts:336]: Add `inherited: boolean`?

File: ui/guest/src/controls/rte.ts [hunk 3]
```diff
@@ -113,7 +114,8 @@ export function initTinyMCE(
         function save() {
           const content = getContent();
           if (changed) {
-            contentController.updateField(record.modelId, field.id, record.index, content);
+            const model = getReferentialEntries({ ...record, fieldId: field.id }).model;
```
Comment (rart) [inline ui/guest/src/controls/rte.ts]: Why not just sent the record? The records field and the field should match.

File: ui/guest/src/react/GuestProxy.tsx [hunk 4]
```diff
@@ -372,7 +373,8 @@ export default function GuestProxy() {
           );
           const model = getCachedModel(modelId);
           const contentType = getCachedContentType(model.craftercms.contentTypeId);
-          const fieldType = ContentType.getField(contentType, fieldId).type;
+          const field = ContentType.getField(contentType, fieldId);
```
Comment (rart) [inline ui/guest/src/react/GuestProxy.tsx]: Not using field later so looks like you can roll this back?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/guest/src/controls/rte.ts (hunk 3 above)
Comment (aachavarria) [reply ui/guest/src/controls/rte.ts]: rollback

File: ui/app/src/services/content.ts [hunk 5]
```diff
@@ -313,7 +313,12 @@ export function moveItem(
...
   // const isTargetEmbedded = nnou(targetParentModelId);
-  if (originalModelId === targetModelId || originalParentModelId === targetParentModelId) {
+  // When Moving between inherited dropzone to other dropzone, the modelsIds will be different but in some cases the parentId will be null for both targets
+  // in that case we need to add a nnou validation to parentsModelId;
+  if (
+    originalModelId === targetModelId ||
+    (nnou(originalParentModelId) && nnou(targetParentModelId) && originalParentModelId === targetParentModelId)
+  ) {
     if (originalParentModelId === targetParentModelId && nnou(originalParentModelId)) {
       debugger;
```
Comment (rart) [inline ui/app/src/services/content.ts]: 👆

File: ui/app/src/utils/model.ts [hunk 6]
```diff
@@ -102,3 +104,17 @@ function extractCollectionPiece(model, fieldId, index) {
     return aux[field];
   }
 }
+
+export function isInheritedField(model, fieldId) {
+  return !!model.craftercms.sourceMap?.[fieldId];
```
Comment (rart) [inline ui/app/src/utils/model.ts]: `!!model.craftercms.sourceMap?.[fieldId]` => `Boolean(model.craftercms.sourceMap?.[fieldId])`

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/content.ts (hunk 5 above)
Comment (aachavarria) [reply ui/app/src/services/content.ts]: removed

**Review by rart — APPROVED** [review]

**Pull request merged**